### PR TITLE
doc: release notes: Update contents for 2.1.0

### PR DESCRIPTION
**Pull request by dleach02**
Add updated issue list and filled in some details from PRs.

Signed-off-by: David Leach <david.leach@nxp.com>

### CODE REVIEW

**Comment by dleach02** [comment]
I have made a pass at filling in details in sections without contributions from owners so please review and provide feedback.

**Review by jhedberg — APPROVED** [review]

**Review by d3zd3z — APPROVED** [review]
Confirm no known security vulnerabilities addressed in this release.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: doc/releases/release-notes-2.1.rst
Comment (dleach02) [reply doc/releases/release-notes-2.1.rst]: will do

**Review by MaureenHelm — APPROVED** [review]

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: doc/releases/release-notes-2.1.rst
Comment (dleach02) [reply doc/releases/release-notes-2.1.rst]: yeah... that was cut and paste from PR. I need to look at that PR a little closer to see if there is a more meaningful phrase

**Review by dbkinder — APPROVED** [review]
Ah, that's better.  Thanks!

**Review by ioannisg — CHANGES_REQUESTED** [review]
For the Architecture release notes, I'd suppose you'd need to put ARCHes in alphabetical order.

File: doc/releases/release-notes-2.1.rst
Comment (jhedberg) [inline doc/releases/release-notes-2.1.rst]: I'd add here "Mesh: Added support for acting as a Provisioner"

**Review by jukkar — APPROVED** [review]

File: doc/releases/release-notes-2.1.rst
Comment (dleach02) [reply doc/releases/release-notes-2.1.rst]: ok

**Comment by dleach02** [comment]
> For the Architecture release notes, I'd suppose you'd need to put ARCHes in alphabetical order.

@ioannisg done

**Review by jhedberg — APPROVED** [review]

**Comment by ioannisg** [comment]
Unless I am mistaken, ARC should be before ARM.

**Comment by dleach02** [comment]
@pgielda, @nategraff-sifive, @ruuddw, @nashif, @andrewboie 

You all have been identified as needing approvals to sign off on the final release notes

**Comment by dleach02** [comment]
> Unless I am mistaken, ARC should be before ARM.

done

**Review by ruuddw — APPROVED** [review]
I'm ok for ARC. For the ARC Architecture section completness, you could add "Updates for SMP support", but it's implicitly covered by issue 19885 in the list below already.

**Review by pgielda — APPROVED** [review]

File: doc/releases/release-notes-2.1.rst
Comment (nashif) [inline doc/releases/release-notes-2.1.rst]: does this really belong here as a major enhancement?

File: doc/releases/release-notes-2.1.rst
Comment (nashif) [inline doc/releases/release-notes-2.1.rst]: I would drop this if there are no other details, this is an internal API, actually this call does not even exist anymore.

File: doc/releases/release-notes-2.1.rst
Comment (nashif) [inline doc/releases/release-notes-2.1.rst]: lets remove that. I will give it another go after this is merged...

**Review by nashif — CHANGES_REQUESTED** [review]

File: doc/releases/release-notes-2.1.rst
Comment (dleach02) [reply doc/releases/release-notes-2.1.rst]: Don't know. Was trying to capture something for this area. I'm completely open to suggestions here.

File: doc/releases/release-notes-2.1.rst
Comment (dleach02) [reply doc/releases/release-notes-2.1.rst]: ok

File: doc/releases/release-notes-2.1.rst
Comment (dleach02) [reply doc/releases/release-notes-2.1.rst]: ok

**Review by nategraff-sifive — APPROVED** [review]
RISC-V LGTM 👍

File: doc/conf.py
Comment (ioannisg) [inline doc/conf.py]: Since you are adding one, remove one from the bottom, so it's the same number of releases

**Review by ioannisg — APPROVED** [review]
Looks ok. I added a minor comment for the number of releases listed in conf.py, @dleach02 please, address this.

**Review by dbkinder — APPROVED** [review]

File: doc/releases/release-notes-2.1.rst
Comment (dleach02) [reply doc/releases/release-notes-2.1.rst]: done

**Review by nashif — APPROVED** [review]

**Pull request merged**
